docs/Compilation/Boogie.md

```diff
@@ -0,0 +1,23 @@
+# Dafny compilation to Boogie
+
+## `{:opaque} attribute`
```

Comment:
I would first explain the effect of the opaque attribute without going into fuel, that the body becomes completely invisible for opaque functions until reveal is called.

I suggest only explaining fuel if you want to explain the `{:fuel ...}` attribute.

Reply:
> Since this is a file that explains how we compile to Boogie

Shouldn't documentation on that be inside the code?

Reply:
Code comments are nice, but I think it's perfectly fine to complement them with out-of-band documentation. Indeed, we do this already for some compiler issues (see `docs/Compilation/*.md`).